shared/src/main/java/com/couchbase/lite/DocumentReplication.java

```diff
@@ -72,62 +72,44 @@ public boolean isPush() {
     /**
      * The current document id.
      */
-    public String getDocumentId() {
+    public String getDocumentID() {
         return docId;
     }
 
     /**
-     * The current document revision id.
-     */
-    public String getRevisionId() {
-        return revId;
-    }
-
-    /**
-     * The current document id.
+     * The current deleted status of the document.
      */
     public boolean isDeleted() {
         return isDeleted;
     }
 
     /**
-     * The current document flags.
+     * The current access removed status of the document.
      */
-    public int getFlags() {
-        return flags;
-    }
+    public boolean isAccessRemoved()  {
+        return isAccessRemoved;
+    } //TODO: will find a way to get this value and write unit test for it in the related PR for the ticket.
 
     /**
      * The current document replication error.
      */
-    public C4Error getError() {
-        return error;
+    public CouchbaseLiteException getError() {
+        return error.getCode() != 0 ? CBLStatus.convertError(error) : null;
```

Comment:
Just want to check if it is possible that the error could null itself? Could you please confirm?

Reply:
Looks like default errorCode value is 0.